### PR TITLE
Onboard TPC-H workload in LST-Bench

### PR DESCRIPTION
**Pull request by jcamachor**
See #106 .

### CODE REVIEW

**Comment by ashvina** [comment]
Hi @jcamachor.
Thanks for the PR. Could you provide additional insight on the rationale behind integrating TPC-H? It seems like a good idea since TPC-H is a popular benchmark for OLAP systems. However, it misses a refresh phase as pointed out in the description of this PR. I am assuming the refresh will be part of an upcoming PR. If you could elaborate on the upcoming enhancement related to refresh phase for TPC-H, that'd be great. I'm also keen to understand which aspects of table formats are expected to get impacted by this TPC-H and its custom refresh phase enhancement.

**Comment by jcamachor** [comment]
Thanks @ashvina. As you mentioned, TPC-H is another widely utilized benchmark for decision systems, but differing slightly from TPC-DS. One notable difference is the finer granularity of refresh phases compared to TPC-DS. Another difference is that in the throughput phase, the refresh stream is executed concurrently with other streams running read queries. 
Introducing the core TPC-H as a base workload opens up interesting directions, such as leveraging it for evaluating engines/tables using a different workload, and also being able to use its task templates for designing slightly modified workloads (e.g., longevity).

Regarding the refresh phase's implementation, it has already been incorporated into the PR (comment in the description was outdated). You can find the code in #106. The implementation closely mirrors the refresh phase used in TPC-DS; thus, we did not need to use any custom task definition as introduced in #105.

File: src/test/resources/config/spark/w_all_tpch-delta.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Description: Workload for test: All task types, TPC-H, Delta
+---
+version: 1
+id: w_all_tpch
```
Comment (anjagruenheid) [inline src/test/resources/config/spark/w_all_tpch-delta.yaml:4]: Should these be w_all_tpch_delta/iceberg/hudi respectively?

File: src/main/resources/scripts/tpch/build/spark/2_load_part.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+INSERT
+    INTO
+        ${catalog}.${database}.part SELECT
```
Comment (anjagruenheid) [inline src/main/resources/scripts/tpch/build/spark/2_load_part.sql:3]: Nit: indentation?

**Review by anjagruenheid — APPROVED** [review]

**Comment by jcamachor** [comment]
Closes #106

**Pull request merged**